annotation/src/main/java/io/smallrye/common/annotation/Experimental.java

```diff
@@ -0,0 +1,24 @@
+package io.smallrye.common.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation that specifies that an element is experimental and may change without notice.
+ */
+@Inherited
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD, ElementType.TYPE })
+public @interface Experimental {
+    /**
+     * Describes why the annotated element is experimental.
+     * 
+     * @return the experimental description.
+     */
+    String value() default "";
```

Comment:
Did you consider not providing a default value, to ensure an actual value is entered by anyone using the annotation?

Reply:
Maybe this was already discussed (and dismissed) but what about having an enum with values like CHANGES_SPECIFICATION, NOT_COVERED_IN_SPECIFICATION, etc? It would enable us to scan the code with tooling to assess how much we've diverged from the spec. An optional String description could be kept too.

Reply:
> Did you consider not providing a default value, to ensure an actual value is entered by anyone using the annotation?

I did. I thought that in some cases, the java docs would be self sufficient to provide the explanation (in case of new classes for instance), and then requiring description in Experimental might be duplicating the information.

I have no issue making it mandatory.

Reply:
> Maybe this was already discussed (and dismissed) but what about having an enum with values like CHANGES_SPECIFICATION, NOT_COVERED_IN_SPECIFICATION, etc? It would enable us to scan the code with tooling to assess how much we've diverged from the spec. An optional String description could be kept too.

Actually, we didn't thought about that. Might be a good idea. We do need to figure which values should we use. Do you want to come with a proposal? Or do you think these are enough?

Reply:
I can send a PR and we'll see what others think. I suppose that if we do this, then the description should stay optional, there's no reason to require both to be specified.

Reply:
Please, do :)

Reply:
I did propose having an enum, but I find it hard to come up with a comprehensive set of values. IMHO, mandatory `String value()` is enough.

Reply:
It was discussed in the policy document.

The outcome was to stick with something flexible for now, just a description, and see if it made sense to constrain it further as we begin to use it

Reply:
> > Did you consider not providing a default value, to ensure an actual value is entered by anyone using the annotation?
> 
> I did. I thought that in some cases, the java docs would be self sufficient to provide the explanation (in case of new classes for instance), and then requiring description in Experimental might be duplicating the information.
> 
> I have no issue making it mandatory.

It would be good to force a description, otherwise it's up to the developer to actually write javadocs.